aws/table_aws_ebs_snapshot.go

```diff
@@ -362,8 +332,9 @@ func buildEbsSnapshotFilter(ctx context.Context, d *plugin.QueryData, h *plugin.
 	if equalQuals["owner_id"] != nil {
 		ownerFilter.Name = aws.String("owner-id")
 		ownerFilter.Values = []string{equalQuals["owner_id"].GetStringValue()}
-	} else {
+	} else if equalQuals["owner_alias"] == nil && equalQuals["snapshot_id"] == nil {
```

Comment:
Please add a comment explaining "why" we have this check in (the "what" can be added to, but the "why" is most important"), and also add a line or two into the table doc explaining the new behaviour, along with adding  any missing examples highlighting how to list private snapshots, shared (from another AWS account) snapshots, and a specific snapshot.